src/frontend/src/catalog.rs

```diff
@@ -65,13 +66,8 @@ pub struct FrontendCatalogManager {
 
 #[async_trait::async_trait]
 impl CacheInvalidator for FrontendCatalogManager {
-    async fn invalidate_table(&self, _ctx: &Context, table_ident: TableIdent) -> MetaResult<()> {
-        let table_id = table_ident.table_id;
-        let key = TableNameKey::new(
-            &table_ident.catalog,
-            &table_ident.schema,
-            &table_ident.table,
-        );
+    async fn invalidate_table_name(&self, _ctx: &Context, table_name: TableName) -> MetaResult<()> {
```

Comment:
I don't think it's a good idea to separate `invalidate_table` into `invalidate_table_name` and `invalidate_table_id`. It incurs cache inconsistency. Especially `TableInfo` and `TableName` should be invalidated at the same time.

Reply:
I think it won't cause any inconsistency if the `TableInfo` does not contain table name. In that case, the `TableInfo` and `TableName` are orthogonal, i.e. they are used separately independently.

Reply:
Trust me, any accidental misuse may lead to bugs 🐛 . Or we should change the `TableInfo::name`'s visibility to pub(crate) to protect potential misuse.

Reply:
I believe in you. However, the bugs are guarded by thoroughly tests. I will remove the catalog, schema and table names in `TableInfo`.